core/src/test/java/hudson/UtilTest.java

```diff
@@ -490,9 +494,13 @@ private void lockFileForDeletion(File f) throws IOException, InterruptedExceptio
         // On unix, can't use "chattr +u" because ext fs ignores it.
         // On Windows, we can't delete files that are open for reading, so we use that.
         assert Functions.isWindows();
-        final InputStream s = new FileInputStream(f);
+        final FileChannel channel = FileChannel.open(f.toPath(), StandardOpenOption.READ, StandardOpenOption.WRITE);
+        final FileLock lock = channel.lock();
```

Comment:
🐛 Such approach may cause the `OverlappingFileLockException ` if the lock is being already held by the JVM. It will cause an unhadled and random Runtime exception here. The best pattern is a loop with tryLock(), but surprisingly it also has to handle this exception.